### PR TITLE
Gp2 208 product search modal

### PR DESCRIPTION
**Pull request by bobmeredith**
To do (delete all that do not apply):

 - [x] Change has a jira ticket that has the correct status.
 - [x] Changelog entry added.
 - [x] (if adding env vars) Added any new environment variable to vault.

### CODE REVIEW

File: config/env/secrets-template [hunk 1]
```diff
@@ -24,4 +24,6 @@ DIRECTORY_FORMS_API_SENDER_ID=debug#DO NOT ADD SECRETS TO THIS FILE
 AIRTABLE_API_KEY=debug#DO NOT ADD SECRETS TO THIS FILE
 #DO NOT ADD SECRETS TO THIS FILE
 DIT_HELPDESK_URL=http://debug.com#DO NOT ADD SECRETS TO THIS FILE
+#DO NOT ADD SECRETS TO THIS FILE
+3CE_COMMODITY_SEARCH_TOKEN=debug#DO NOT ADD SECRETS TO THIS FILE
```
Comment (depsiatwal) [inline config/env/secrets-template:28]: probably not ideal to name variables with a number

File: config/settings.py [hunk 2]
```diff
@@ -431,6 +431,11 @@
     'MADB_URL', 'https://www.check-duties-customs-exporting-goods.service.gov.uk'
 )
 
+# 3CE commodity classification
+COMMODITY_SEARCH_URL = 'http://info.dev.3ceonline.com/ccce/apis/classify/v1/interactive/classify-start'
+COMMODITY_SEARCH_REFINE_URL = 'http://info.dev.3ceonline.com/ccce/apis/classify/v1/interactive/classify-continue'
```
Comment (depsiatwal) [inline config/settings.py:436]: Should this not be set as an env variable ?

**Review by lewis-coulson-dit — APPROVED** [review]

File: config/settings.py (hunk 2 above)
Comment (bobmeredith) [reply config/settings.py:436]: Good question. I thought about this, but reached the conclusion that hard-coding is better on the basis that 3ce don't provide distinct develop/live endpoints (rather they use the bearer token to determine access) and thus the urls would be the same in all environments.  If the urls were fed in from vault, it adds more moving parts, and we would need to set the same URLs the same in all environments.
I only just noticed the 'dev' in the URL, but these URLs are printed in their API documentation .pdf so there's really only these.
Also, took the lead from the MAPDB url above.

File: config/env/secrets-template (hunk 1 above)
Comment (bobmeredith) [reply config/env/secrets-template:28]: Good point.  TBH, it didn't cross my mind.  How important do you think this is to change?

File: config/settings.py (hunk 2 above)
Comment (webbyfox) [reply config/settings.py:436]: Bob, still a good idea to keep this in env to minimise further development in future and keeping the same practise and align with other APIs, what do you think?

**Review by webbyfox — COMMENTED** [review]
Apart from the above comments its looks really good

File: react-components/src/components/ProductFinder/CountryFinder.jsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    modalContent.style.maxHeight='700px'
+  }
+
+  function saveCountry(country) {
+    setSelectedCountry(country);
+    let result = Services.updateExportPlan({export_countries:[country]}).then((result) => {
+      closeModal();
+      // window.location.reload(true);   # do we need to reload page
+    }).catch((result) => {
+      debugger;
```
Comment (dahfool) [inline react-components/src/components/ProductFinder/CountryFinder.jsx:48]: Please remove debugger and code commented out.

File: config/settings.py (hunk 2 above)
Comment (depsiatwal) [reply config/settings.py:436]: Agree with Rizwan about consistently. Also if URLS changed then we we would require a release 
may be low risk for a single URL but we all external sources at one point one of them change domains/urls. 
We can get around setting in all environments by setting a default in settings and only add to Vault to override.

File: config/env/secrets-template (hunk 1 above)
Comment (depsiatwal) [reply config/env/secrets-template:28]: I think either we allow this or we don't 
https://www.w3schools.com/python/python_variables.asp says it's bad :-) 
surprised flake8 didn't pick it up

**Review by depsiatwal — APPROVED** [review]
LGTM

File: config/env/secrets-template (hunk 1 above)
Comment (bobmeredith) [reply config/env/secrets-template]: Yes, have changed to CCCE instead.

File: config/settings.py (hunk 2 above)
Comment (bobmeredith) [reply config/settings.py]: NP, will set as env variables with a default in settings. We won't need to define in Vault unless the urls change for some reason.

File: react-components/src/components/ProductFinder/CountryFinder.jsx (hunk 3 above)
Comment (bobmeredith) [reply react-components/src/components/ProductFinder/CountryFinder.jsx]: Will do.  Nothing to put in api catches yet though - I'll add  TODOs and do whatever the design says when there's a design.

**Pull request merged**